python/core/qgseditformconfig.sip

```diff
@@ -145,15 +145,6 @@ class QgsEditFormConfig
     /**
      * Get the id for the editor widget used to represent the field at the given index
      *
-     * @param fieldIdx  The index of the field
-     *
-     * @return The id for the editor widget or a NULL string if not applicable
-     */
-    QString widgetType( int fieldIdx ) const;
-
-    /**
-     * Get the id for the editor widget used to represent the field at the given index
-     *
```

Comment:
I think we should go all or nothing here. Either we allow indexed access for all properties or for none.

So I'd say also replace `removeWidgetConfig(idx)` and others or leave the old methods in place.